api/src/test/java/org/pmiops/workbench/workspaces/WorkspacesControllerTest.java

```diff
@@ -828,68 +832,102 @@ public void testListForApproval() throws Exception {
   public void testCloneWorkspace() throws Exception {
     stubFcGetGroup();
     stubFcGetWorkspaceACL();
-    Workspace workspace = createWorkspace();
-    workspace = workspacesController.createWorkspace(workspace).getBody();
+    Workspace originalWorkspace = createWorkspace();
+    originalWorkspace = workspacesController.createWorkspace(originalWorkspace).getBody();
 
     // The original workspace is shared with one other user.
-    DbUser writerUser = new DbUser();
-    writerUser.setEmail("writerfriend@gmail.com");
-    writerUser.setUserId(124L);
-    writerUser.setDisabled(false);
-
-    writerUser = userDao.save(writerUser);
+    DbUser writerUser = createAndSaveUser("writerfriend@gmail.com", 124L);
     ShareWorkspaceRequest shareWorkspaceRequest = new ShareWorkspaceRequest();
-    shareWorkspaceRequest.setWorkspaceEtag(workspace.getEtag());
-    UserRole creator = new UserRole();
-    creator.setEmail(LOGGED_IN_USER_EMAIL);
-    creator.setRole(WorkspaceAccessLevel.OWNER);
-    shareWorkspaceRequest.addItemsItem(creator);
-    UserRole writer = new UserRole();
-    writer.setEmail(writerUser.getEmail());
-    writer.setRole(WorkspaceAccessLevel.WRITER);
-    shareWorkspaceRequest.addItemsItem(writer);
+    shareWorkspaceRequest.setWorkspaceEtag(originalWorkspace.getEtag());
+
+    createUserToShareWith(shareWorkspaceRequest, LOGGED_IN_USER_EMAIL, WorkspaceAccessLevel.OWNER);
+
+    createUserToShareWith(
+        shareWorkspaceRequest, writerUser.getEmail(), WorkspaceAccessLevel.WRITER);
 
     stubFcUpdateWorkspaceACL();
     workspacesController.shareWorkspace(
-        workspace.getNamespace(), workspace.getName(), shareWorkspaceRequest);
+        originalWorkspace.getNamespace(), originalWorkspace.getName(), shareWorkspaceRequest);
 
-    CloneWorkspaceRequest req = new CloneWorkspaceRequest();
-    Workspace modWorkspace = new Workspace();
+    final Workspace modWorkspace = new Workspace();
     modWorkspace.setName("cloned");
     modWorkspace.setNamespace("cloned-ns");
-    ResearchPurpose modPurpose = new ResearchPurpose();
+
+    final ResearchPurpose modPurpose = new ResearchPurpose();
     modPurpose.setAncestry(true);
     modWorkspace.setResearchPurpose(modPurpose);
```

Comment:
I'm not following. The research purpose is already the last thing set on `modWorkspace`.

Reply:
```
final ResearchPurpose modPurpose = new ResearchPurpose();
modPurpose.setAncestry(true);
modWorkspace.setResearchPurpose(modPurpose);
modPurpose.setPopulation(true);
modPurpose.setPopulationDetails(...);
```

I found it slightly confusing that we set the research purpose on the workspace and then continued to modify the research purpose.

Ofc, this works since we're using references but on a quick pass, I assume that objects are fully constructed when they're being set to their parent.

Ex. 
```
final ResearchPurpose modPurpose = new ResearchPurpose();
modPurpose.setAncestry(true);
modPurpose.setPopulation(true);
modPurpose.setPopulationDetails(...);

modWorkspace.setResearchPurpose(modPurpose);
```